### DISCUSSION TITLE
Some skinning and gameplay QoL features to make the game just a little bit flashier

### DISCUSSION THREAD

**Discussion by SIUUUUUUUUUU17** [ppy/osu#29901] [Ideas]
**1. 3D Lane design**
This can be used to mimic the design seen in many mobile rhythm games. The "angle" of the playfield can be adjusted in skin.ini, between 0 and 60 degrees, with 0 the classic vertical playfield and 60 the maximum "angle", mimicking the looks of games such as Project SEKAI or Arcaea
![Example using Sonolus](https://github.com/user-attachments/assets/aac86a68-6604-45b2-a020-1932ac6d166d)

**2. Option to set hitsounds per mode**
An option to use separate hitsounds for certain modes could be implemented into skin.ini, similar to how Mania implements separate textures for different key counts. Useful for skin makers who want to keep hitsounds but prefers to remove them or set a singular one on specific modes, like Mania for example

**3. Combo counter customizations**
One can customize how the combo counter is presented, changing its animation ease and duration, whether it counts gradually or instantly like the Triangles or Classic counter, texture effect for every number increase, etc...

**4. Use hit effects as judgement**
Similar to games like VOEZ, hit effects can be used to indicate their judgement (PERFECT, GREAT, etc...). Textures can be configured (i.e. making multiple versions of a texture, then adding them to skin.ini) to do this

**5. Late/Early indicators**
This allows players to know if they hit the notes late or early so they can adjust their offset more efficiently. The amount of times either of these indicators appear in gameplay are shown at the result cards except for Catch clears. I prefer these skinnable per mode

**6. Show cover image as a HUD element**
Arcaea does this with its song jackets, so why not? Aspect ratio can be configured in the HUD editor

**7. Ability to limit character amount for the BeatmapAttributesText HUD element**
Some song titles are absurdly long (i.e. pretty much every TUYU song titles), so having this would be nice

**Comment by peppy** [comment]
Most of this is planned!

**Comment by TechniKris** [comment]
I like your ideas a lot, especially 3D lane and early/late indicators.
- **Early/Late** could be solved by adding e.g. `hit100early.png`/`hit50early.png`, if present in a skin, they could be used for early hits and the regular `hit100.png`/`hit50.png` for late hits.
- **3D lane** is much easier to read IMO (I'm not a mania player, but I have lots of experience with Guitar Hero etc.) so I guess it'd be an unranked Fun mod, but it'd be really fun to play

**Reply by Tenexxt** [reply]
If I could play with a 3D lane on mania, i could actually come back to it after months.
